### PR TITLE
PWGHF: DpH Corr check Gen Flag

### PR DESCRIPTION
**Pull request by apalasciano**
At gen level, considering only correlations between D->\pi K \pi and other charged particles considering the flagMcMatchGen flag

### CODE REVIEW

File: PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx
Comment (vkucera) [inline PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:527]: Use `TESTBIT`

File: PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx
Comment (vkucera) [inline PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:420]:
```suggestion
        isDplusSignal = TESTBIT(std::abs(candidate.flagMcMatchRec()), aod::hf_cand_3prong::DecayType::DplusToPiKPi);
```

Comment (apalasciano) [reply PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx]: If there are no major issues, it is crucial to have this PR in today's tag since it's needed for HP approvals

**Comment by apalasciano** [comment]
If there are no major issues, it is crucial to have this PR in today's tag since it's needed for HP approvals

**Review by fcolamar — APPROVED** [review]
Ciao Antonio, I see, approving now.
Cheers,
Fabio

**Pull request merged**
